packages/web3-eth-contract/src/contract.ts

```diff
@@ -1053,7 +1053,7 @@ export class Contract<Abi extends ContractAbi>
 		const transactionToSend = sendTransaction(this, tx, DEFAULT_RETURN_FORMAT);
 
 		// eslint-disable-next-line no-void
-		void transactionToSend.on('contractExecutionError', (error: unknown) => {
+		void transactionToSend.on('error', (error: unknown) => {
```

Comment:
More details below, but I removed `contractExecutionError` event emission from `sendTransaction` and instead used the existing `error` event that emits an instance of `ContractExecutionError`